ggshield/utils/git_shell.py

```diff
@@ -234,6 +234,11 @@ def git_ls(wd: Optional[Union[str, Path]] = None) -> List[str]:
     return git(cmd, timeout=600, cwd=wd).split("\n")
 
 
+def git_ls_unstaged(wd: Optional[Union[str, Path]] = None) -> List[str]:
+    cmd = ["ls-files", "--other", "--exclude-standard"]
```

Comment:
As you said, this command does not work with submodules. This is a problem because users are going to believe we scan unknown files inside submodules, giving them a false sense of security. I am afraid we have to implement submodules recursion ourselves.

Reply:
This has been a bit more tricky than expected, but it should work as expected now. I'm more worried about the robustness of the code now.